addons/Coloring/src/presenter.js

```diff
@@ -71,12 +71,16 @@ function AddonColoring_create(){
     function setColorsThatCanBeFilled() {
         var configuration = presenter.configuration;
         configuration.colorsThatCanBeFilled = [];
-        $.each(configuration.areas, function() {
-            var color = presenter.getColorAtPoint(this.x, this.y);
+        const wholeCanvas = presenter.ctx.getImageData(0, 0, presenter.canvasWidth, presenter.canvasHeight).data;
+        const areas = configuration.areas;
+
+        for (var i = areas.length; i--;) {
+            let area = areas[i];
```

Comment:
słowo kluczowe let nie powinno być zastąpione var ?

Reply:
Czemu? For tworzy scope, var nie jest block scoped, a let tak. Nie potrzebujemy dostępu do zmiennej area poza konkretnym użyciem w tej jednej iteracji pętli.